frontend/device/pocketbook/device.lua

```diff
@@ -154,6 +168,44 @@ function PocketBook:setDateTime(year, month, day, hour, min, sec)
     end
 end
 
+-- Predicate, so no self
+function PocketBook.canAssociateFileExtensions()
+    local f = io.open(ext_path, "r")
+    if not f then return true end
+    local l = f:read("*line")
+    f:close()
+    if l and not l:match("^#koreader") then
+        return false
+    end
+    return true
+end
+
+function PocketBook:associateFileExtensions(assoc)
+    -- First load the system-wide table, from which we'll snoop file types and icons
+    local info = {}
+    for l in io.lines("/ebrmain/config/extensions.cfg") do
+        local m = { l:match("^([^:]*):([^:]*):([^:]*):([^:]*):(.*)") }
```

Comment:
Out of curiosity, what does that file look like?

Reply:
Vanilla file on 6.x firmwares is:

```
m3u:@M3U_file:3:music_player.app:ICON_MP3
mp3:@MP3_file:3:music_player.app:ICON_MP3
hrss:@HTML_file:1:browser.app:ICON_HTM
mhtml:@MHTML_file:1:browser.app:ICON_MHTML
jpg:@JPEG_image:2:picviewer.app:ICON_JPG
jpeg:@JPEG_image:2:picviewer.app:ICON_JPG
png:@PNG_image:2:picviewer.app:ICON_PNG
bmp:@BMP_image:2:picviewer.app:ICON_BMP
tif:@TIFF_image:2:picviewer.app:ICON_TIFF
tiff:@TIFF_image:2:picviewer.app:ICON_TIFF
file:@MENCA_file:1:menca_downloader.app:ICON_MENCA
txt:@Text_file:1:eink-reader.app:ICON_TXT
fb2:@FB2_file:1:eink-reader.app:ICON_FB2
htm:@HTML_file:1:eink-reader.app:ICON_HTM
html:@HTML_file:1:eink-reader.app,browser.app:ICON_HTML
chm:@Z_HTML_file:1:eink-reader.app:ICON_CHM
rtf:@RTF_file:1:eink-reader.app:ICON_RTF
doc:@DOC_file:1:eink-reader.app:ICON_DOC
docx:@DOC_file:1:eink-reader.app:ICON_DOCX
docm:@DOC_file:1:eink-reader.app:ICON_DOC
prc:@PRC_file:1:eink-reader.app:ICON_PRC
pdb:@PRC_file:1:eink-reader.app:ICON_PRC
mobi:@PRC_file:1:eink-reader.app:ICON_PRC
tcr:@TCR_file:1:eink-reader.app:ICON_TCR
pdf:@PDF_file:1:eink-reader_with_pdfium.app,eink-reader_with_rmsdk.app:ICON_PDF
epub:@EPUB_file:1:eink-reader_with_blink.app,eink-reader_with_epub2.app:ICON_EPUB
acsm:@ACSM_file:1:eink-reader.app:ICON_ACSM
djvu:@DJVU_file:1:eink-reader.app:ICON_DJVU
djv:@DJVU_file:1:eink-reader.app:ICON_DJVU
cbr:@CBR_file:1:eink-reader.app:ICON_CBR
cbz:@CBZ_file:1:eink-reader.app:ICON_CBZ
app:@APP_file:4:start.app:ICON_APP
run:@APP_file:4:start.app:ICON_APP
lndrm:@LNDRM_file:1:libronet_switcher.app:ICON_LNDRM
pbi:@APP_file:1:sysinstall.app:ICON_APP
pn:@PN_file:1:pb_file_handler:ICON_PN
```